sdk/aqueduct/artifacts/base_artifact.py

```diff
@@ -41,6 +44,12 @@ def set_operator_type(self, operator_type: OperatorType) -> None:
     def set_name(self, name: str) -> None:
         self._dag.update_artifact_name(self._artifact_id, sanitize_artifact_name(name))
 
+    def enable_snapshot(self) -> None:
+        self._dag.update_artifact_should_persist(self._artifact_id, True)
+
+    def disable_snapshot(self) -> None:
```

Comment:
Can we use the `set_` prefix for any write methods?

Reply:
It's been discussed in https://aqueducthq.slack.com/archives/C01U6RREC7M/p1685489357239879 and given it's this close to release, I'd stick with the decision than making more back-and-forth to the channel. We've been using other verbs like `bound` `max` over artifacts so I don't see it as a huge misalignment as long as they start with verbs

Reply:
oh I didn't realize this was user-facing actually. That's good for now yea